overrides/main.html

```diff
@@ -0,0 +1,21 @@
+src: "https://js.monitor.azure.com/scripts/b/ai.3.gbl.min.js",
+name: "appsec.appInsights",
+ld: 0,
+// useXhr: 1,
+crossOrigin: "anonymous",
+// onInit: null,
+// cr: 0,
+cfg: { // Application Insights Configuration
+ connectionString: "InstrumentationKey=57276352-88e8-44a2-8475-7983d748f4a6;IngestionEndpoint=https://northeurope-2.in.applicationinsights.azure.com/;LiveEndpoint=https://northeurope.livediagnostics.monitor.azure.com/",
```

Comment:
Should we expose the instrumentationKey on the html file?
Would it be possible to have it in github secrets and fetch in on deployment?

Reply:
I think this is part of the design when the metrics are collected from the front end. The instrumentation key needs to be available to the front-end in order for it to send the mertrics to the proper recipient. It could leas to potential "spamming" - that's part of the threat model.

So i don't think storing instrumentation key in a secret would increase secuirty since it will be served to the browser clients anyway.

Reply:
One "downside" in having it in this _public repository_ is that "anyone" can now start pushing data into here.
But I guess the site itself is public, so there would not really be a difference?

Reply:
No, there would be no difference as far as I can see. (Besides that different audience and solutions can/will visit either|or the public site or the public repo - the result would potentially be the same)

Reply:
It's good to know that it has been addressed during the threat modeling of the application.
I think as far as the risk is known and accepted it should be ok.
Maybe look into rate limiting for app insights, in case someone tries to DoS?

Reply:
I am considering collecting "similar" metrics from the back-end. The metrics would most likely serve out needs since we are mostly looking for page views and not browser metrics like page view, but we have to change the hosting model and move away from the AZ Static web app to platforms like Raidx (which we should do as a next step)